applications/GeoMechanicsApplication/custom_conditions/U_Pw_normal_face_load_condition.cpp

```diff
@@ -92,22 +86,23 @@ void UPwNormalFaceLoadCondition<TDim, TNumNodes>::CalculateTractionVector(array_
                                                                           const NormalFaceLoadVariables& Variables,
                                                                           const unsigned int& GPoint)
 {
-    Vector NormalVector = ZeroVector(TDim);
-    double NormalStress = MathUtils<>::Dot(row(NContainer, GPoint), Variables.NormalStressVector);
+    const auto normal_stress = MathUtils<>::Dot(row(NContainer, GPoint), Variables.NormalStressVector);
 
+    Vector normal_vector = ZeroVector(3);
     if constexpr (TDim == 2) {
-        const double TangentialStress =
+        const auto tangential_stress =
             MathUtils<>::Dot(row(NContainer, GPoint), Variables.TangentialStressVector);
-        NormalVector       = column(Jacobian, 0);
-        rTractionVector[0] = TangentialStress * NormalVector[0] - NormalStress * NormalVector[1];
-        rTractionVector[1] = NormalStress * NormalVector[0] + TangentialStress * NormalVector[1];
-    } else if constexpr (TDim == 3) {
-        // Since the normal vector is pointing outwards for the 3D conditions, the normal stress
-        // should switch sign, such that positive normal contact stress is defined inwards.
-        NormalStress *= -1;
 
-        MathUtils<double>::CrossProduct(NormalVector, column(Jacobian, 0), column(Jacobian, 1));
-        rTractionVector = NormalStress * NormalVector;
+        Vector tangential_vector = ZeroVector(3);
+        std::copy_n(column(Jacobian, 0).begin(), TDim, tangential_vector.begin());
+        Vector out_of_plane_vector = ZeroVector(3);
+        out_of_plane_vector[2]     = 1.0;
+        MathUtils<double>::CrossProduct(normal_vector, out_of_plane_vector, tangential_vector);
```

Comment:
The template parameter has a default value of `double`, so you don't need to specify it:
```suggestion
        MathUtils<>::CrossProduct(normal_vector, out_of_plane_vector, tangential_vector);
```

Reply:
done, also on 104